### ISSUE TITLE
FreeCAD does not launch: "Fatal Error compiling builtin function"

### ISSUE THREAD

**Issue by jorgenatz** [void-linux/void-packages#12595]
### System

* xuname:  
  Void 4.19.48_1 x86_64-musl GenuineIntel notuptodate rF
* package:  
  freecad

### Expected behavior
When 
`$  FreeCAD`
is typed, I had expected the program to start as normal, showing a splash screen until subsequently opening a window.

### Actual behavior
When running, this error is produced:
```
Error compiling builtin: 
Fatal error compiling builtin function 'apply': Program received signal SIGSEGV, Segmentation fault.
sorry, no backtrace on musl libc
```
The startup splash screen is shown with a variety of intialization messages before the crash.

This is the entire output:
```
$ FreeCAD
FreeCAD 0.18.2, Libs: 0.18.2R1_voidlinux
© Juergen Riegel, Werner Mayer, Yorik van Havre 2001-2019
  #####                 ####  ###   ####  
  #                    #      # #   #   # 
  #     ##  #### ####  #     #   #  #   # 
  ####  # # #  # #  #  #     #####  #   # 
  #     #   #### ####  #    #     # #   # 
  #     #   #    #     #    #     # #   #  ##  ##  ##
  #     #   #### ####   ### #     # ####   ##  ##  ##

Error compiling builtin: 
Fatal error compiling builtin function 'apply': Program received signal SIGSEGV, Segmentation fault.
sorry, no backtrace on musl libc
```
### Steps to reproduce the behavior
Simply launch FreeCAD from a terminal or graphical menu.

It seems that this may be linked to QtWebKit when linked with musl, as indicated by this similar issue: [https://github.com/annulen/webkit/issues/708](https://github.com/annulen/webkit/issues/708)

If I have forgotten to include anything, please let me know, and thank you for reading.

**Comment by pullmoll** [comment]
The bottom comment in the linked similar issue is probably what causes the crash.

We always have trouble with libraries and applications wanting to use huge portions of the stack for local data, yet failing to set their stack space requirements with `pthread_attr_setstacksize(3)` in their thread(s). This applies for Musl libc with its 80KiB default stack size, not glibc, which has a huge default stack size. As a consequence developers tend to don't care because of this.

qt5-webkit is old and most probably needs (a lot of?) patches to correctly define its requirements. If we are going to patch it that can only be guesswork and trial+error, with the main issue being to identify the places in the code where it should happen in the first place.

**Comment by pullmoll** [comment]
Perhaps you can try to build qt5-webkit on your own after adding
```
LDFLAGS="-Wl,-z,stack-size=2097152"
```
in the `srcpkgs/qt5-webkit/template` e.g. after the line `CXXFLAGS=...`.
If that works for you, we can update qt5-webkit for everyone's benefit.

**Comment by yopito** [comment]
I saw this bug too while packaging FreeCAD 0.18 (related to qt).

@pullmoll:  change stack size already been tried (issue #8909) without success, but without rebuilding qt. I've tried `sudo chelf -s 82099200 /usr/bin/FreeCAD` for instance.

One possible alternate workaround: disable the startWorkBench on startup in the user's setting file:

```
$ diff -bu ~/.FreeCAD/ORIG/user.cfg ~/.FreeCAD/user.cfg 
--- .FreeCAD/ORIG/user.cfg      2019-03-03 13:28:58.546387911 +0100
+++ .FreeCAD/user.cfg   2019-03-03 14:10:19.245254436 +0100
@@ -9,7 +9,7 @@
	   <FCParamGroup Name="Preferences">
		 <FCParamGroup Name="General">
		   <FCText Name="FileOpenSavePath">/home/yopito</FCText>
-          <FCText Name="AutoloadModule">StartWorkbench</FCText></FCParamGroup>
+          <FCText Name="AutoloadModule">NoneWorkbench</FCText></FCParamGroup>
		 <FCParamGroup Name="Units"/>
		 <FCParamGroup Name="Macro"/>
		 <FCParamGroup Name="Mod">
```
This way, FreeCAD is launched on musl. However, I don't know if it's sufficient for real-word usage, if FreeCAD uses qt5-webkit functionalities somewhere else.

**Comment by Johnnynator** [comment]
We could remove qt5-webkit dependency on musl for FreeCAD, so the potential unstable parts won't even be build for musl. (If this is possible)

**Comment by Gottox** [comment]
Does it help if you run the following?
```
chelf -s 2097152 /usr/bin/FreeCAD
```

**Comment by yopito** [comment]
thanks for suggestion. I give a try on FreeCAD without qt5-webkit, and also see if outgoing qt 5.13 packaging could help here too.

**Comment by yopito** [comment]
Here some test results on x86_64/musl (Everything is fine with glibc)

1. FreeCAD 0.18.2_1  (from VoidRepo, not rebuilt) with qt5-webkit 5.12 updated (PR #13212):
	* still crash with qt5-webkit.  gdb's backtrace is really similar to old one  
	(https://gist.github.com/yopito/3408054baf3626f6a167d6521dfa2d50)
	* changing stack size up to 80 MB does not fix it (`sudo chelf -s 82099200 /usr/bin/FreeCAD`)

2. FreeCAD 0.18.2 rebuilt without qt5-webkit-devel:
	* build is fine
	* basic runtime is fine
	* "Start" Workbench can't be loaded as expected ("`WebGUI not found`")
	* Test mode (`FreeCAD -t 0`) segfaults 

3. FreeCAD 0.18.2_1  (from VoidRepo, not rebuilt) with qt-webkit 5.12 (updated, see PR #13212) rebuilt with specific stacksize (see pullmoll's post above): still fails as point 1, no difference

At this point, I have no other option to update FreeCAD, in order to built without qt5-webkit support on musl

**Comment by yopito** [comment]
@jorgenatz: package freecad 0.18.2_2 is available without qt5-webkit build dep for musl.  
Could you give it a try and feeback on its usablity with "real" usage ?  
Then close this PR if it's fine for you. thx.
